### PR TITLE
store tests: encode some invariants into prisms, add more tests

### PR DESCRIPTION
**Pull request by christian-marie**

### CODE REVIEW

File: lib/Network/OAuth2/Server/Types.hs [hunk 1]
```diff
@@ -157,7 +160,8 @@ instance ToField UserID where
     toField = toField . unpackUserID
 
 userid :: Prism' ByteString UserID
-userid = prism' (T.encodeUtf8 . unpackUserID) (Just . UserID . T.decodeUtf8)
+userid = prism' (T.encodeUtf8 . unpackUserID)
+                (fmap UserID . hush . T.decodeUtf8')
```
Comment (christian-marie) [inline lib/Network/OAuth2/Server/Types.hs]: Subtle but important change here

Comment (thsutton) [inline lib/Network/OAuth2/Server/Types.hs]: This is just `Nothing` instead of throwing an exception, right?

Comment (oswynb) [inline lib/Network/OAuth2/Server/Types.hs]: Yis

**Comment by oswynb** [comment]
This looks the good to me, shall merge post-rebase.

**Pull request merged**